protocol/proofs/new-chains.md

```diff
@@ -0,0 +1,70 @@
+This is required because the L2 genesis state references a number of L1 contracts, including the `SystemConfig` and
+`OptimismPortal`. The fault proof system creates two cyclic dependencies:
+
+1. The `faultGameGenesisOutputRoot` is used by the L1 `AnchorStateRegistry` contract as the initial anchor state for
+   permissioned and permissionless games. For new chains it must be set to the L2 genesis output root, which is
+   unknowable prior to deploing the L1 contracts.
+2. The `faultGameAbsolutePrestate` is used by the L1 `FaultDisputeGame` contract to define the version of `op-program`
+   to be used as part of permissioned and permissionless dispute games. `op-program` requires the L2 chain
+   configuration (rollup config and L2 genesis state) to be available in the version of the superchain-registry that it
+   includes to operate correctly. Similar to above, the L2 chain configuration is unknowable prior to deploying the L1
+   contracts. Additionally, chains cannot be added to the superchain-registry until _after_ they have launched.
+
+# Proposed Solution
+
+TBD
+
+# Alternatives Considered
+
+## Defer Deployment of Dispute Game Types
+
+Remove the `FaultDisputeGame` and its dependent contracts (including `AnchorStateRegistry`) from the initial L1
+deployment. The `DisputeGameFactory` would still be deployment, but with no supported game types. It would be impossible
+to create any dispute games and thus withdrawals would be disabled. The `faultGameGenesisOutputRoot` and
+`faultGameAbsolutePrestate` values would not be required ahead of this initial L1 deployment as they are only used by
+the dispute game implementations.
+
+The L2 genesis state could then be created based on the deployed L1 contracts. Optionally the chain could be started at
+this point without support for withdrawals.
+
+The config would need to be added to a version of `op-program` either by adding the chain to the superchain-registry,
+updating the superchain-registry dependency in the optimism monorepo and tagging a new release of `op-program` or by
+creating some kind of custom build of `op-program`.
+
+Finally, the `ProxyAdminOwner` would deploy the `FaultDisputeGame` and dependent contracts, providing the correct
+`faultGameGenesisOutputRoot` and `faultGameAbsolutePrestate` values and call `setImplementation` on the
+`DisputeGameFactory`. Withdrawals would then be possible.
+
+### Pros
+
+* Neatly avoids the cyclic dependency.
+* Chains that are not supporting permissionless games yet could deploy only the permissioned game, preventing creating
+  permissionless games entirely.
+
+### Cons
+
+* The `ProxyAdminOwner` is required to perform the `setImplementation` calls because ownership of the L1 contracts has
+  already been transferred. In the current system, the L1 contract deployment can be performed by anyone and the system
+  is fully configured (albeit incorrectly for proofs) before ownership is transferred.
```

Comment:
This is a big downside, because we want the ability to deploy chains with the ProxyAdminOwner configured as the 2/2, meaning the Security Council needs to send a transaction for every new chain

Perhaps you could spin this as a positive, as it means new chains cannot permissionlessly give their keys to the SC, and this lets the `setImplementation` call as a formal way to signify acceptance of governing the chain. (Though they still can permissionlessly give their keys to SC by deploying themselves to v1.3.0 (MCP), then upgrading to FPs))

Reply:
I don't think it changes anything about being able to permissionlessly assign. You could deploy the whole system assigning ownership to yourself, run it for a few years and then change the owner to the SC without permission as I understand it.

A possible work around for this is to not reassign ownership in the first transaction that deploys. So you deploy the L1 contracts except the dispute game implementations, retaining ownership, generate the L2 genesis, deploy the game implementations and then assign the final owner. Not having atomic assignment of ownership does open up some possible security implications (if the deployer's key is compromised or if the deployer messes with things before assigning) but the deployer could just mess with the deployment script now.  With OPSM it would be different but OPSM could keep ownership for itself and be able to finish off the deployment in a trusted way I think (though not sure how it would verify the L2 genesis root....)